.github/workflows/ruby-core-test-suite.yml

```diff
@@ -0,0 +1,52 @@
+name: ruby-core-test
```

Comment:
I would call the workflows and the job simply "ruby-core", and "ruby_core" respectively!